gym/envs/registration.py

```diff
@@ -621,30 +621,37 @@ def make(
     mode = _kwargs.get("render_mode")
     apply_human_rendering = False
 
-    # If we have access to metadata we check that "render_mode" is valid
-    if hasattr(env_creator, "metadata"):
-        render_modes = env_creator.metadata["render_modes"]
-
-        # We might be able to fall back to the HumanRendering wrapper if 'human' rendering is not supported natively
-        if (
-            mode == "human"
-            and "human" not in render_modes
-            and ("single_rgb_array" in render_modes or "rgb_array" in render_modes)
-        ):
-            logger.warn(
-                "You are trying to use 'human' rendering for an environment that doesn't natively support it. "
-                "The HumanRendering wrapper is being applied to your environment."
-            )
-            _kwargs["render_mode"] = (
-                "single_rgb_array"
-                if "single_rgb_array" in env_creator.metadata["render_modes"]
-                else "rgb_array"
-            )
-            apply_human_rendering = True
-        elif mode is not None and mode not in render_modes:
-            raise error.Error(
-                f"Invalid render_mode provided: {mode}. Valid render_modes: None, {', '.join(render_modes)}"
-            )
+    # If we have access to metadata we check that "render_mode" is valid and see if the HumanRendering wrapper needs to be applied
+    if mode is not None:
+        if hasattr(env_creator, "metadata"):
+            assert isinstance(
+                env_creator.metadata, dict
+            ), f"Expect the environment creator ({env_creator}) metadata to be dict, actual type: {type(env_creator.metadata)}"
+
+            if "render_modes" in env_creator.metadata:
+                render_modes = env_creator.metadata["render_modes"]
+
+                # Apply the `HumanRendering` wrapper, if the mode=="human" but "human" not in render_modes
+                if (
+                    mode == "human"
+                    and "human" not in render_modes
+                    and (
+                        "single_rgb_array" in render_modes
+                        or "rgb_array" in render_modes
+                    )
+                ):
+                    logger.warn(
+                        "You are trying to use 'human' rendering for an environment that doesn't natively support it. "
+                        "The HumanRendering wrapper is being applied to your environment."
+                    )
+                    apply_human_rendering = True
+                    if "single_rgb_array" in render_modes:
+                        _kwargs["render_mode"] = "single_rgb_array"
+                    else:
+                        _kwargs["render_mode"] = "rgb_array"
+                # else:
+                #   we don't raise an error as the environment may have forgotten to add the "render_modes"
+                #   if the render mode is not valid for an environment, the environment should raise an error
```

Comment:
Maybe one of my last comments was somewhat unclear. I think that it is fair to give a warning if `env_creator` has metadata that doesn't contain "render_modes". My point was that we shouldn't worry if `env_creator` has no metadata at all because there may be legitimate reasons for that. Similarly, I think it is fair to give a warning if metadata exists and has "render_modes", but the current mode isn't contained in that list.